platform/security/sts/security-sts-propertyclaimshandler/src/main/java/org/codice/ddf/security/sts/claims/property/AttributeFileClaimsHandler.java

```diff
@@ -148,16 +130,51 @@ public ProcessedClaimCollection retrieveClaimValues(
   }
 
   @Override
-  public List<String> getSupportedRealms() {
-    return supportedRealms;
+  public List<Serializable> getValues(final String attributeName) {
+    return Optional.ofNullable(systemHighAttributes.get(attributeName))
+        .orElse(Collections.emptyList());
   }
 
-  @Override
-  public String getHandlerRealm() {
-    return realm;
+  public void setAttributeFileLocation(String attributeFileLocation) throws IOException {
+    this.attributeFileLocation =
+        notNull(attributeFileLocation, "attributeFileLocation may not be null");
+    init();
   }
 
-  public void setAttributeFileLocation(String attributeFileLocation) {
-    this.attributeFileLocation = attributeFileLocation;
+  private void setSupportedClaimTypes() {
+    Set<Map.Entry<String, Object>> entries = json.entrySet();
+    Set<String> claims = new HashSet<>();
+    for (Map.Entry<String, Object> entry : entries) {
+      Object value = entry.getValue();
+      if (value instanceof Map) {
+        Set keySet = ((Map) value).keySet();
+        for (Object key : keySet) {
+          claims.add((String) key);
+        }
+      }
+    }
+
+    supportedClaimTypes = claims.stream().map(URI::create).collect(Collectors.toList());
```

Comment:
❗️ `supportedClaimTypes` needs to be `volatile` to make sure it's not being changed while `getSupportedClaimTypes()` is called.